simulation/types/account.go

```diff
@@ -95,6 +95,43 @@ func (sim *SimCtx) SelAddrWithDenoms(ctx sdk.Context, denoms []string) (simulati
 	return acc, balance, true
 }
 
+// SelAddrWithDenom attempts to find an address with the provided denom. This function
+// returns (account, randSubsetCoins, found), so if found = false, then no such address exists.
+// randSubsetCoins is a random subset of the provided denoms, if the account is found.
+// TODO: Write unit test
+func (sim *SimCtx) SelAddrWithDenom(ctx sdk.Context, denom string) (simulation.Account, sdk.Coin, bool) {
+	denomSlice := []string{denom}
+	accHasDenoms := func(acc simulation.Account) bool {
+		return !sim.App.GetBankKeeper().GetBalance(ctx, acc.Address, denom).Amount.IsZero()
+	}
+
+	acc, accExists := sim.RandomSimAccountWithConstraint(accHasDenoms)
+	if !accExists {
+		return acc, sdk.Coin{}, false
+	}
+	balance := sim.RandCoinSubset(ctx, acc.Address, denomSlice)
+	balanceCoin := balance[0]
+	return acc, balanceCoin, true
+}
+
+// RandomSimAccountWithKDenoms returns a random account that possesses greater than or equal to the requested(k) denoms
+// TODO: Write unit test
+func (sim *SimCtx) RandomSimAccountWithKDenoms(ctx sdk.Context, k int) (simulation.Account, sdk.Coins, bool) {
+	var coins sdk.Coins
+	accHasBal := func(acc simulation.Account) bool {
+		if len(sim.App.GetBankKeeper().SpendableCoins(ctx, acc.Address)) >= k {
+			return true
+		}
+		coins = sim.App.GetBankKeeper().SpendableCoins(ctx, acc.Address)
+		return false
+	}
+	acc, found := sim.RandomSimAccountWithConstraint(accHasBal)
+	if !found {
+		return simulation.Account{}, sdk.Coins{}, false
+	}
+	return acc, coins, true
+}
```

Comment:
I recommend we make this function not return coins, and then just become: 
```
func (sim *SimCtx) RandomSimAccountWithKDenoms(ctx sdk.Context, k int) (simulation.Account, sdk.Coins, bool) {
	accHasBal := func(acc simulation.Account) bool {
		return len(sim.App.GetBankKeeper().SpendableCoins(ctx, acc.Address)) >= k 
	}
	return sim.RandomSimAccountWithConstraint(accHasBal)
}
```

After looking at the API consequences / this in more detail, I think the previous API I made for JoinPool, which combined getting the subset of tokens with getting the account, was a bad design.

Reply:
Then we make a second function with thats `GetRandSubsetOfKDenoms(ctx, addr, k)`

Reply:
@ValarDragon was this what you were suggesting? https://github.com/osmosis-labs/osmosis/pull/2091/commits/64c385fad5956521c42778e9ffa7fd10af24f8cd

Side note but still an issue, for some reason when running these tests, no gamm or token factory results are showing in the end report. Did we disable this in one of the merges?